### PR TITLE
Specify data-diff arguments using config files

### PR DESCRIPTION
**Pull request by erezsh**
Addresses issue #82

### CODE REVIEW

File: data_diff/config.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+        run_args[f"database{index}"] = database
+        run_args[f"table{index}"] = table
+        if threads is not None:
+            run_args[f"threads{index}"] = int(threads)
+
+    # Update keywords
+    new_kw = dict(kw)  # Set defaults
+    new_kw.update(run_args)  # Apply config
+    new_kw.update({k: v for k, v in kw.items() if v})  # Apply non-empty defaults
```
Comment (nolar) [inline data_diff/config.py:60]: I am not sure how this `if v` works here: If the new value of `verbose` is False (e.g., due to a hypothetical `--quiet` flag), will it override the TOML setting or be ignored? The same for values that are int 0, if any. Would it make more sense only to exclude `if v is not None` — i.e. the special marker of emptiness?

File: README.md [hunk 2]
```diff
@@ -201,6 +201,57 @@ Options:
...
+
+### How to use with a configuration file
+
+Data-diff lets you load the configuration for a run from a TOML file.
+
+Reasons to use a configuration file:
+
+- Convenience - Set-up the parameters for diffs that need to run often
+
+- Easier and more readible - You can define the database connection settings as config values, instead of in a URI.
```
[the remark this answers is not in the document]
Comment (erezsh) [reply README.md:214]: The config values are usually in a URI, as in, the URI contains them.

File: README.md [hunk 3]
```diff
@@ -201,6 +201,57 @@ Options:
...
+# Specify the connection params to the test database.
+[database.test_postgresql]
+driver = "postgresql"
+user = "postgres"
+password = "Password1"
+
+# Specify the default run params
+[run.default]
+update_column = "timestamp"
+verbose = true
```
[the remark this answers is not in the document]
Comment (erezsh) [reply README.md:236]: I'm not simulating a list structure. I'm using `run` and `database` as a prefix, to denote category.

When you collect each run together into a list, then they are runs.

How does it help YAML? I don't understand.

File: data_diff/config.py (hunk 1 above)
Comment (erezsh) [reply data_diff/config.py:60]: The default for `click` is False, not None. I don't control the default values, I just define it as `is_flag`.

Yes, for a `--quiet` flag I'll have to change the way it works. Either get rid of `click`, or more likely, define the parameters in our own format, and translate that for click.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (nolar) [reply README.md:236]: I meant the config readability — the same story as e.g. with the REST endpoints being plural by convention:

```yaml
runs:
  default: ...
  some-other: ...
  ...
databases:
  test_postgresql: ...
  test_mysql: ...
  ...
```

Maybe I misunderstood the syntax of TOML — from the code below, it looks like they are parsed into lists/dicts (i.e. "containers") with the dot-name ("run.…", "database.…") unfolded into a nested structure.

Let's keep it as it is then.

File: data_diff/config.py (hunk 1 above)
Comment (nolar) [reply data_diff/config.py:60]: I see. It makes sense now: so, the default False should not override the config value. Thanks for clarifying.

**Review by nolar — APPROVED** [review]

**Pull request merged**